### PR TITLE
Perform infrastructure upgrades

### PR DESCRIPTION
**Pull request by lavelle**
- Node 16
- Yarn 3
- TypeScript

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -65,7 +65,7 @@ jobs:
                   key: ${{ runner.os }}-${{ hashFiles('**/yarn.lock') }}
             - name: Install dependencies
               run: |
-                  yarn install --frozen-lockfile
+                  yarn install
```
Comment (danoc) [inline .github/workflows/test.yml:68]:
```suggestion
                  yarn install --immutable
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -46,16 +46,16 @@ jobs:
                   yarn install --frozen-lockfile
             - name: Run ESLint
               run: |
-                  yarn netlify-zip-it-ship-it-issue-38-workaround && yarn start && yarn run test:eslint
```
Comment (danoc) [inline .github/workflows/test.yml:49]: Hah, I guess the very verbose name that I chose for this Yarn script ended up coming in handy. Looks like the issue is fixed.

File: .gitignore [hunk 3]
```diff
@@ -2,7 +2,7 @@
 # https://github.com/netlify/zip-it-and-ship-it/issues/38
```
Comment (danoc) [inline .gitignore:2]: Even though the issue is fixed, I imagine we still want to keep these around since developers might have the file in their local environments?

File: netlify.toml [hunk 4]
```diff
@@ -1,5 +1,5 @@
 [build]
-  command = "yarn install --frozen-lockfile --prod && yarn run netlify-zip-it-ship-it-issue-38-workaround"
+  command = "yarn install"
```
Comment (danoc) [inline netlify.toml:2]:
```suggestion
  command = "yarn install --immutable"
```

File: package.json [hunk 5]
```diff
@@ -55,12 +64,17 @@
     },
     "homepage": "https://thumbprint.design/tokens",
     "dependencies": {
+        "@netlify/functions": "^1.0.0",
         "apollo-server-lambda": "^2.9.7",
         "graphql": "^14.5.8"
     },
+    "resolutions": {
+        "@types/istanbul-lib-report": "3.0.0"
```
Comment (danoc) [inline package.json:72]: Why is this needed?

File: package.json [hunk 6]
```diff
@@ -55,12 +64,17 @@
     },
     "homepage": "https://thumbprint.design/tokens",
     "dependencies": {
+        "@netlify/functions": "^1.0.0",
```
Comment (danoc) [inline package.json:67]: What changed that makes this needed?

File: src/__snapshots__/index.test.ts.snap [hunk 7]
```diff
@@ -66,6 +66,9 @@ exports[`outputted Android matches snapshot 1`] = `
...
...low500 = exports.tpColorYellow = exports.tpColorYellow300 = exports.tpColorYellow200 = exports.tpColorYellow100 = exports.tpColorGreen600 = exports.tpColorGreen500 = exports.tpColorGreen = exports.tpColorGreen300 = exports.tpColorGreen200 = exports.tpColorGreen100 = exports.tpColorPurple600 = exports.tpColorPurple500 = exports.tpColorPurple = exports.tpColorPurple300 = exports.tpColorPurple200 = exports.tpColorPurple100 = exports.tpColorIndigo600 = exports.tpColorIndigo500 = exports.tpColorIndigo = exports.tpColorIndigo300 = exports.tpColorIndigo200 = exports.tpColorIndigo100 = exports.tpColorBlue600 = exports.tpColorBlue500 = exports.tpColorBlue = exports.tpColorBlue300 = exports.tpColorBlue200 = exports.tpColorBlue100 = exports.tpBreakpointSplitViewSmallValue = exports.tpBreakpointSplitViewSmall = exports.tpBreakpointSplitViewMediumValue = exports.tpBreakpointSplitViewMedium = exports.tpBreakpointSmallValue = exports.tpBreakpointSmall = exports.tpBreakpointMediumValue = exports.tpBreakpointMedium = exports.tpBreakpointLargeValue = exports.tpBreakpointLarge = exports.tpBorderRadiusSides = exports.tpBorderRadiusFull = exports.tpBorderRadiusBig = exports.tpBorderRadiusBase = void 0;
```
Comment (danoc) [inline src/__snapshots__/index.test.ts.snap:69]: This seems incorrect, I think?

File: package.json (hunk 6 above)
Comment (lavelle) [reply package.json:67]: i Converted the netlify function to TypeScript and this provides the types.

File: package.json (hunk 5 above)
Comment (lavelle) [reply package.json:72]: I had the same issue in the icons conversion. It's a transitive dependency of TS Jest, and it causes the script to fail with a type error if you use the default version it ships with.

File: .gitignore (hunk 3 above)
Comment (lavelle) [reply .gitignore:2]: Yeah that was my thinking, leave it for now

File: src/__snapshots__/index.test.ts.snap (hunk 7 above)
Comment (lavelle) [reply src/__snapshots__/index.test.ts.snap:69]: hmm actually i think this is just an artifact of the TS version upgrade. all these values are still assigned correctly and exported below. it just also has this boilerplate of initialising them up here.

Comment (danoc) [reply src/__snapshots__/index.test.ts.snap:69]: Weird!

**Review by danoc — APPROVED** [review]
